src/app/chart/sparkline/sparkline.component.ts

```diff
@@ -0,0 +1,224 @@
+import { Component, Input, OnInit, DoCheck } from '@angular/core';
+
+import { cloneDeep, defaults, isEqual, merge } from 'lodash';
+import { ChartDefaults } from '../chart.defaults';
+import { ChartBase } from '../chart.base';
+import { SparklineConfig } from './sparkline-config';
+
+/**
+ * Sparkline chart component based on C3
+ */
+@Component({
+  selector: 'pfng-chart-sparkline',
+  templateUrl: './sparkline.component.html'
+})
+export class SparklineComponent extends ChartBase implements OnInit, DoCheck {
+  /**
+   * Chart data for the chart
+   */
+  @Input() chartData: any;
+
+  /**
+   * Configuration object containing details about how to render the chart
+   */
+  @Input() config: SparklineConfig;
+  private prevChartData: any;
+  public sparklineChartId: any;
+  private defaultConfig: SparklineConfig;
+
+  /**
+   * Default constructor
+   * @param chartDefaults
+   */
+  constructor(private chartDefaults: ChartDefaults) {
+    super();
+  }
+
+  /**
+   * NgOnInit implementation
+   */
+  ngOnInit(): void {
+    // Create an ID for the chart based on the chartId in the config if given
+    if (this.sparklineChartId === undefined) {
+      this.sparklineChartId = 'sparklineChart';
+      if (this.config.chartId) {
+        this.sparklineChartId = this.config.chartId + this.sparklineChartId;
+      }
+    }
+
+    this.setupConfig();
+    this.updateAll();
+    super.generateChart(this.sparklineChartId);
+  }
+
+  /**
+   *  Check if the component config has changed
+   */
+  ngDoCheck(): void {
+    // Do a deep compare on config
+    if (!isEqual(this.chartData, this.prevChartData)) {
+      this.updateAll();
+    }
+
+    if (!isEqual(this.config, this.prevConfig)) {
```

Comment:
prevConfig appears to be only updated upon ngOnInit when  super.generateChart is called.

Do we want to call super.ngOnInit here? Otherwise, maybe we should compare prevChartData to see if that has changed -- prevConfig probably won't change after ngOnInit, right?